math/wxMaxima/Portfile

```diff
@@ -1,112 +1,78 @@
-    }
-
-    livecheck.type  none
-}
+homepage            https://github.com/wxMaxima-developers/wxmaxima
+long_description    ${description}: version ${version}
 
-if {${subport} eq ${name}} {
-    conflicts       wxMaxima-devel
-    long_description ${description}: \
-                    Provides the release version, which is typically updated every four months.
-
-    # get the source tarball from github.
-    github.setup    andrejv wxmaxima 18.02.0 Version-
-    checksums       rmd160  d1017d56bfd76e9c51dcc3230be89ebca1724560 \
-                    sha256  77f0414be54d8ea15acedfc3cedf72a57cc2b3ba5d4835a53f3009e050c05483 \
-                    size    10131293
-    revision        1
-
-    patchfiles      patch-src_Dirstructure.cpp.diff
-    post-patch {
-        reinplace -W ${worksrcpath} "s|@@PREFIX@@|${prefix}|g" \
-                                    src/Dirstructure.cpp
-    }
-    
-    livecheck.regex Version-(\[a-z0-9.\]+)${extract.suffix}
-}
+github.setup        wxMaxima-developers wxmaxima ${version} Version-
+checksums           rmd160  0c10602f6f1d777ff16fe85500f57222eb07fc27 \
+                    sha256  055ff34eb98539716a18f7cc11861b5a790fde46aaafabb68f1e690cb649d9df \
+                    size    15314478
 
-universal_variant   no
+livecheck.regex     Version-(\[a-z0-9.\]+)${extract.suffix}
 
-variant wx32 description {Use wxWidgets 3.2 - experimental!} {}
+universal_variant   no
 
-if {[variant_exists wx32] && [variant_isset wx32]} {
-    # we want to use the wxWidgets-3.2
-    wxWidgets.use   wxWidgets-3.2
-} else {
-    wxWidgets.use   wxWidgets-3.0
-}
+# Note: as of Match 2020 (1b00466)the build with wxWidgets-3.2 has serious issues with scrolling
+wxWidgets.use       wxWidgets-3.0
 
-depends_lib         port:${wxWidgets.port}
+depends_lib         port:${wxWidgets.port} port:libomp-devel
 
-depends_build       path:bin/cmake:cmake
+depends_build       port:cmake
 
-depends_run         bin:maxima:maxima
+depends_run         port:maxima port:gnuplot
 
 patch.pre_args      -Np0
 
 use_configure       no
 
+post-patch {
+    reinplace -W ${worksrcpath} "s|OSX_MACPORTS_PREFIX  *\"${prefix}\"|OSX_MACPORTS_PREFIX \"${prefix}\" // patched by MacPorts|" src/Dirstructure.cpp
+    reinplace -W ${worksrcpath} "s|OSX_MACPORTS_PREFER 0|OSX_MACPORTS_PREFER 1 // patched by MacPorts|" src/Dirstructure.cpp
+}
+
 build {
     system -W ${worksrcpath}       "mkdir ./build"
-    system -W ${worksrcpath}/build "cmake -DwxWidgets_CONFIG_EXECUTABLE:FILEPATH=${wxWidgets.wxconfig} .."
+    system -W ${worksrcpath}/build "cmake -DwxWidgets_CONFIG_EXECUTABLE:FILEPATH=${wxWidgets.wxconfig} -DUSE_CPPCHECK=YES -DUSE_OPENMP=YES -DCMAKE_CXX_FLAGS=-I${prefix}/include/libomp -DOpenMP_CXX_FLAGS='-Xpreprocessor -fopenmp' -DOpenMP_CXX_LIB_NAMES=libomp -DOpenMP_libomp_LIBRARY=${prefix}/lib/libomp/libomp.dylib .."
+    system -W ${worksrcpath}/build "cmake -LA ."
     system -W ${worksrcpath}/build "cmake --build ."
 }
```

Comment:
Since it uses `cmake` to build, have you tried using the cmake 1.1 portgroup instead of trying to manually build here?

Reply:
No, I didn't try this. I guess the lengthy cmake configure phase command line could be given in some way? Do you have a reference project using the camke portgroup you can recommend - of cause I can search for one but I am not sure all have the same quality, so if you have one at hand you can recommend as good example, it might save us one iteration.

Reply:
> Actually I did remove a variant for the wxWidgets version, which I thought would be more damaging to maintenance than helpful.

Oh I see that now. Sorry I missed it. So please adjust the commit message to mention removing the -devel sub-port.

Reply:
You can get a list by typing looking at [this track ticket](https://trac.macports.org/ticket/47197#comment:78). Some of the ports there use the older cmake 1.0 portgroup however. We want to move every port to the 1.1 version. 

I would look at `libssh` as an example. It uses the cmake 1.1 portgroup and adds some custom cmake configure arguments.

Reply:
> Since it uses `cmake` to build, have you tried using the cmake 1.1 portgroup instead of trying to manually build here?

I tried it and it doesn't work for me. It might be useful for building libraries and console apps, but building application packages like wxMaxima is a different level of complexity. When I faithfully convert my settings to using the portgroup it does build successfully, but there is no application package in the build tree (searching for it with find) and destroot fails. So I will revert to what I have and leave it for the future to debug the cmake port group (I already spend a good week getting it to work as is without breaking wxMaxima CI and wxMaxima builds on other platforms).

Reply:
Thank you for trying. I'm not familiar enough with cmake to offer suggestions and I don't have time to look into it myself anyway.